cert_manager/__init__.py

```diff
@@ -121,7 +121,7 @@ def request_missing(self, configured_domains, current_domains):
         for main_domain, domain_group in configured_domains.items():
             # Check if at least some of the domains are not covered by the current certificate
             domains = domain_group['domains']
-            if set(domains) - current_domains.get(main_domain, set()):
+            if set(domains) != set(current_domains.get(main_domain, [])):
```

Comment:
@mtyaka this condition statement is a bit confusing...

As a person who is reading this code for the first time and based on what I understood from the testing instructions, the `main_domain` is either the `internal_lms_domain` or the `external_lms_domain` in case it is available.

Meanwhile, the `domains = domain_group['domains']` seems to be the list of domains that are available in the certificate, especially since they are later on in the code requested using `self.request_cert(domains)`.

So,`set(domains)` seems like it will never be equal to `set(current_domains.get(main_domain, []))`, unless both were empty...

Could you please provide some clarification about that, if you think no code change is necessary?

Reply:
@nizarmah

> the `main_domain` is either the `internal_lms_domain` or the `external_lms_domain` in case it is available.

That is correct. The value of `main_domain` is also used in the filename of the resulting certificate. Each `domain_group` corresponds to a separate certificate.

> Meanwhile, the `domains = domain_group['domains']` seems to be the list of domains that are available in the certificate

In this method, that is the list of the domains that [Ocim stores into consul KV](https://github.com/open-craft/opencraft/blob/27debc438fe9c3eb35738733427c308d6c371d98/instance/models/openedx_instance.py#L553-L562) for a particular instance. This tries to determine whether the list of domains in the existing certificate (if it exists at all) is equal to the list of domains in consul. If it isn't, it requests a new certificate with the correct list of domains. Note that `domain_group['domains']` also includes `main_domain` (which is not obvious from the code).

> So,`set(domains)` seems like it will never be equal to `set(current_domains.get(main_domain, []))`, unless both were empty...

The names of variables are not very good. `configured_domains` means domains as they are configured inside Consul KV (by Ocim) and `current_domains` is the list of the domains actually present in currently existing certificate files.

Why do you think they will never be euqal?

Reply:
The reason behind me not thinking they'll be equal is that I expected `domains` to be a list which is usually larger than 1 (5 internal domains, at least). Meanwhile, it was clear that `main_domain` is of size 1.

But then I found [these lines](https://github.com/open-craft/cert-manager/blob/f386abea1d99f0ae760672f3da670de24c6398c1/cert_manager/__init__.py#L149-L153), which are:
```
                configured_domains[domain] = {
                    'domains': [domain],
                    # Assume DNS entries for additional have been set up a long time ago:
                    'dns_records_updated': 0,
                }
```

They made it clear that each `domains` in a `domain_group` is going to contain a single domain. 

So yeah, I can see now how they can be equal 👍🏼 

Sorry about that, I should have done more digging before adding the comment. 
But hey, I'm starting to gain more insight on the infrastructure, which is nice! 😄